dependencies.yaml

```diff
@@ -124,32 +124,44 @@ dependencies:
         matrices:
           - matrix:
               cuda: "11.8"
-              arch: x86_64
             packages:
               - cudatoolkit=11.8
-              - libcufile=1.4.0.31
-              - libcufile-dev=1.4.0.31
+          - matrix:
+              cuda: "11.5"
+            packages:
+              - cudatoolkit=11.5
+          - matrix:
+              cuda: "11.4"
+            packages:
+              - cudatoolkit=11.4
+          - matrix:
+              cuda: "11.2"
+            packages:
+              - cudatoolkit=11.2
+      - output_types: conda
```

Comment:
I don't understand the schema, but it looks like the `output_types: conda` key is repeated here?

Reply:
The reason this structure is better is that the cudatoolkit dependency isn't arch-specific. We were just using arch information to ensure that libcufile was only installed on x86_64. However, this will change at some point when libcufile adds aarch64 support. Separating these now ensures that we can cleanly define the dependencies with smaller changes in the future.

Reply:
The other important piece to know is that the first matching "matrix" combination is selected, meaning that we cannot just write it in a single list (simplified to 11.8/11.5 for demonstration):

```yaml
  # This is not enough because only the first matching matrix element can be selected
  cudatoolkit:
    specific:
      - output_types: conda
        matrices:
          - matrix:
              cuda: "11.8"
            packages:
              - cudatoolkit=11.8
          - matrix:
              cuda: "11.5"
            packages:
              - cudatoolkit=11.5
          - matrix:
              cuda: "11.8"
              arch: x86_64
            packages:
              # libcufile package version reference: https://anaconda.org/nvidia/libcufile/files
              - libcufile=1.4.0.31
              - libcufile-dev=1.4.0.31
          - matrix:
              cuda: "11.5"
              arch: x86_64
            packages:
              - libcufile>=1.1.0.37,<=1.1.1.25
              - libcufile-dev>=1.1.0.37,<=1.1.1.25
```

```yaml
  # This is a viable alternative based on the "first matching" criterion but it repeats
  # cudatoolkit dependencies in two package lists, which is not ideal
  cudatoolkit:
    specific:
      - output_types: conda
        matrices:
          - matrix:
              cuda: "11.8"
              arch: x86_64
            packages:
              - cudatoolkit=11.8
              # libcufile package version reference: https://anaconda.org/nvidia/libcufile/files
              - libcufile=1.4.0.31
              - libcufile-dev=1.4.0.31
          - matrix:
              cuda: "11.8"
            packages:
              - cudatoolkit=11.8
          - matrix:
              cuda: "11.5"
              arch: x86_64
            packages:
              - cudatoolkit=11.5
              - libcufile>=1.1.0.37,<=1.1.1.25
              - libcufile-dev>=1.1.0.37,<=1.1.1.25
          - matrix:
              cuda: "11.5"
            packages:
              - cudatoolkit=11.5
```

Reply:
Thanks for the explanation!